server/src/main/java/mateusz/michal/chat/Registration/Service/RegistrationService.java

```diff
@@ -187,34 +175,42 @@ private boolean isUserInDatabaseByEmail(String email){
         return user.isPresent();
     }
 
-    private boolean isEmailIncorrect(String email){
+    private boolean isEmailCorrect(String email){
         Pattern pattern = Pattern.compile("[a-zA-Z]+[\\.a-zA-Z0-9]*@[a-zA-Z0-9]+\\.[a-z]{2,}[a-z]*");
-        return !pattern.matcher(email).matches();
+        return pattern.matcher(email).matches();
     }
 
-    private boolean isPasswordIncorrect(String password) {
+    private boolean isPasswordStrong(String password) {
         Pattern pattern = Pattern.compile("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])" +
-                "(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*])(?!=.*\\s).{8,15}");
-        return !pattern.matcher(password).matches();
+                "(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*]).{0,}");
```

Comment:
Coś nie pasuje mi w tym regexpie, nie znam się na regexpach Javy, ale zazwyczaj by to znaczyło:
`[\\!\\@\\#]` -> znak z grupy: `\\` - znak `\`; `!` - znak `!`; `\\` - (znowu) znak `\`; `@` - znak `@` itd...

Czy źle to czytam, czy w Javie regexpy aż tak się różnią?
Plus po co sklejanie dwóch stringów w jeden? Jednak regexp łatwiej się czyta nawet jako jeden długi string :)

Reply:
Rozłożę to na czynniki pierwsze: 
(?=.*\\d) - standardowy positive lookahead który sprawdza czy w stringu występuję przynajmniej jedna cyfra.
(?=.*[a-z]) - sprawdza czy w stringu występuje przynajmniej jedna mała litera.
(?=.*[A-Z]) - sprawdza czy w stringu występuje przynajmniej jedna duza litera.
(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*]) - sprawdza czy w stringu występuje przynajmniej jeden ze znaków specjalnych !-*
Ponieważ w Javie wyrażenia regularne zapisujemy jako stringi dlatego przed znakami specjalnymi umieszczam dwa znaki \\. 
Regexa złamałem zapewne dlatego, że prawdopodobnie przekroczył on maksymalną liczbę znaków która wyświetlała się na monitorze. Sprawdzę to jeszcze potem.

Reply:
Sprawdziłem - mieści się w jednej linijce. 
Poprawiłem.

Reply:
ok, tak też podejrzewałem że po prostu to typowo Javove z podwójnymi `/`